stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -1786,7 +1790,7 @@ reevaluate_if_taskgroup_has_results:;
     }
     // Put the waiting task at the beginning of the wait queue.
     SWIFT_TASK_GROUP_DEBUG_LOG(this, "WATCH OUT, SET WAITER ONTO waitQueue.head = %p", waitQueue.load(std::memory_order_relaxed));
-    if (waitQueue.compare_exchange_strong(
+    if (waitQueue.compare_exchange_weak(
```

Comment:
If you are changing this to weak, you need to put this into a loop - otherwise you may spuriously fail even if the compare of the CAS is successful.

Reply:
I am pretty sure this is in a loop already.

Reply:
This is an 'if', not a while. If you're relying on the outer loop, you need to make sure that you are not having non-idempotent effects